# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaTrackGraph.h
# Commit: 0ae30eae800d
# Full Hash: 0ae30eae800dcad73d997f9f12bb56eb537ef86c
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2020-11-06 04:14:34
# Regressor Bug: 1651745
# File Overlap Count: 1
# Description:
#   Bug 1651745 - Don't buffer more than necessary in AudioInputProcessing. r=padenot
#   
#   When opening a second input track, there will already be some data from its
#   first instantiation in the driver's scratch buffer. If we ignore this data, we
#   end up buffering too much in AudioInputProcessing::Pull and tripping an assert.
# ==============================================================================

diff -r a00ad7266f54 -r 0ae30eae800d dom/media/MediaTrackGraph.h
--- a/dom/media/MediaTrackGraph.h	Thu Nov 05 16:43:24 2020 +0000
+++ b/dom/media/MediaTrackGraph.h	Thu Nov 05 16:44:00 2020 +0000
@@ -127,7 +127,8 @@
    */
   virtual void NotifyInputData(MediaTrackGraphImpl* aGraph,
                                const AudioDataValue* aBuffer, size_t aFrames,
-                               TrackRate aRate, uint32_t aChannels) = 0;
+                               TrackRate aRate, uint32_t aChannels,
+                               uint32_t aAlreadyBuffered) = 0;
 
   /**
    * Number of audio input channels.